lifetime issue when itering over the `sheet_names()`

Here is the code:
``` rust
    let mut xl = Excel::open(&sce).unwrap();
    for stn in xl.sheet_names().unwrap() {
        let range = xl.worksheet_range(stn).unwrap();
        write_range(&mut dest, range).unwrap();
    }
```
I want to iter over the sheets with sheet names. But when I compile it, I got lifetime issue:

```
error[E0499]: cannot borrow `xl` as mutable more than once at a time
  --> src/main.rs:28:21
   |
27 |     for stn in xl.sheet_names().unwrap() {
   |                -- first mutable borrow occurs here
28 |         let range = xl.worksheet_range(stn).unwrap();
   |                     ^^ second mutable borrow occurs here
29 |         write_range(&mut dest, range).unwrap();
30 |     }
   |     - first borrow ends here

error: aborting due to previous error
```

Is there any way to fix it easily?

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -273,9 +273,9 @@ impl Excel {
     /// let mut workbook = Excel::open(path).unwrap();
     /// println!("Sheets: {:#?}", workbook.sheet_names());
     /// ```
-    pub fn sheet_names(&mut self) -> Result<Vec<&str>> {
+    pub fn sheet_names(&mut self) -> Result<Vec<String>> {
         self.initialize()?;
-        Ok(self.sheets.iter().map(|&(ref k, _)| &**k).collect())
+        Ok(self.sheets.iter().map(|&(ref k, _)| k.to_string()).collect())
     }
 }
 

